rpc/types_executables.go

```diff
@@ -0,0 +1,362 @@
+package rpc
+
+import (
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+type CasmCompiledContractClass struct {
+	EntryPointsByType CasmEntryPointsByType `json:"entry_points_by_type"`
+	ByteCode          []*felt.Felt          `json:"bytecode"`
+	Prime             []*felt.Felt          `json:"prime"`
+	CompilerVersion   string                `json:"compiler_version"`
+	Hints             Hints                 `json:"hints"`
+	// a list of sizes of segments in the bytecode, each segment is hashed invidually when computing the bytecode hash
+	BytecodeSegmentLengths []int `json:"bytecode_segment_lengths,omitempty"`
+}
+
+type CasmEntryPointsByType struct {
+	Constructor []CasmEntryPoint `json:"CONSTRUCTOR"`
+	External    []CasmEntryPoint `json:"EXTERNAL"`
+	L1Handler   []CasmEntryPoint `json:"L1_HANDLER"`
+}
+
+type CasmEntryPoint struct {
+	DeprecatedCairoEntryPoint
+	// the hash of the right child
+	Builtin []string `json:"builtins"`
+}
+
+// 2-tuple of pc value and an array of hints to execute, but adapted to a golang struct
+type Hints struct {
+	Int     int
+	HintArr []Hint
+}
+
+func (hints *Hints) Values() (int, []Hint) {
+	return hints.Int, hints.HintArr
+}
+
+func (hints *Hints) Tuple() [2]any {
+	return [2]any{hints.Int, hints.HintArr}
+}
```

Comment:
What are these functions for?

Reply:
https://github.com/NethermindEth/starknet.go/pull/642#discussion_r1824423182

Answered here